tests/components/rova/test_config_flow.py

```diff
@@ -0,0 +1,224 @@
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={
+            CONF_ZIP_CODE: ZIP_CODE,
+            CONF_HOUSE_NUMBER: HOUSE_NUMBER,
+            CONF_HOUSE_NUMBER_SUFFIX: HOUSE_NUMBER_SUFFIX,
+        },
+    )
+    assert result.get("type") == data_entry_flow.FlowResultType.CREATE_ENTRY
+
+    data = result.get("data")
+    assert data
+    assert data[CONF_ZIP_CODE] == ZIP_CODE
+    assert data[CONF_HOUSE_NUMBER] == HOUSE_NUMBER
+    assert data[CONF_HOUSE_NUMBER_SUFFIX] == HOUSE_NUMBER_SUFFIX
+
+
+async def test_abort_if_not_rova_area(
+    hass: HomeAssistant, mock_rova: MagicMock
+) -> None:
+    """Test we abort if rova does not collect at the given address."""
+
+    # test with area where rova does not collect
+    mock_rova.return_value.is_rova_area.return_value = False
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={
+            CONF_ZIP_CODE: ZIP_CODE,
+            CONF_HOUSE_NUMBER: HOUSE_NUMBER,
+            CONF_HOUSE_NUMBER_SUFFIX: HOUSE_NUMBER_SUFFIX,
+        },
+    )
+
+    assert result.get("type") == data_entry_flow.FlowResultType.FORM
+    assert result.get("errors") == {"base": "invalid_rova_area"}
+
+
+async def test_abort_if_already_setup(hass: HomeAssistant) -> None:
+    """Test we abort if rova is already setup."""
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=f"{ZIP_CODE}{HOUSE_NUMBER}{HOUSE_NUMBER_SUFFIX}".strip(),
+        data={
+            CONF_ZIP_CODE: ZIP_CODE,
+            CONF_HOUSE_NUMBER: HOUSE_NUMBER,
+            CONF_HOUSE_NUMBER_SUFFIX: HOUSE_NUMBER_SUFFIX,
+        },
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={
+            CONF_ZIP_CODE: ZIP_CODE,
+            CONF_HOUSE_NUMBER: HOUSE_NUMBER,
+            CONF_HOUSE_NUMBER_SUFFIX: HOUSE_NUMBER_SUFFIX,
+        },
+    )
+    assert result["type"] == data_entry_flow.FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+@pytest.mark.parametrize(
+    ("exception", "error"),
+    [
+        (ConnectTimeout(), "cannot_connect"),
+        (HTTPError(), "cannot_connect"),
+    ],
+)
+async def test_asserts(
+    hass: HomeAssistant, exception: Exception, error: str, mock_rova: MagicMock
+) -> None:
+    """Test the _site_in_configuration_exists method."""
+
+    # test with exception
+    mock_rova.return_value.is_rova_area.side_effect = exception
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={
+            CONF_ZIP_CODE: ZIP_CODE,
+            CONF_HOUSE_NUMBER: HOUSE_NUMBER,
+            CONF_HOUSE_NUMBER_SUFFIX: HOUSE_NUMBER_SUFFIX,
+        },
+    )
+    assert result.get("type") == data_entry_flow.FlowResultType.FORM
+    assert result.get("errors") == {"base": error}
```

Comment:
This test now only needs to repatch the lib to not raise the side effect and to show that the integration is able to set up after an error